src/Core/Managed/Shared/Channel/InMemoryChannel.cs

```diff
@@ -133,7 +143,14 @@ public void Flush()
         /// <param name="timeout">Timeout interval to abort sending.</param>
         public void Flush(TimeSpan timeout)
         {
-            this.transmitter.Flush(timeout);
+            if (!isDisposed)
+            {
+                this.transmitter.Flush(timeout);
+            }
+            else
+            {
+                throw new ObjectDisposedException("InMemoryChannel", "The telemetry channel has been disposed.");
```

Comment:
@dnduffy Thanks for fixing the problem in the `InMemoryTransmitter`.
@SergeyKanzhelev I think we need you to make the decision regarding changes in the `InMemoryChannel`. I still maintain that this change should not be made because it
- Is not related to the bug #364 this pull request is meant to address
- Breaks the exception behavior contract established by the previous implementation of the `InMemoryChannel`
- Can cause user data loss by throwing unnecessary `ObjectDisposedException` when `Flush` is called for an `InMemoryChannel` already disposed by an application framework without user knowledge. 
- Unnecessarily couples `InMemoryChannel` to `InMemoryTransmitter` by over-specifying assumptions of how one class should work in another